src/hooks/useForkRef/useForkRef.ts

```diff
@@ -0,0 +1,30 @@
+import { type Ref, type RefCallback, type RefObject, useMemo } from 'react'
+
+const setRef = <T>(
+  ref: RefObject<T | null> | ((instance: T | null) => void) | null | undefined,
+  value: T | null,
+) => {
+  if (typeof ref === 'function') ref(value)
+  else if (ref) ref.current = value
+}
+
+/**
+ * Sets multiple refs with a single handler.
+ * @param refs - The refs to set.
+ * @returns A single ref handler that sets all the provided refs which can be passed to the `ref` prop of a component.
+ */
+export const useForkRef = <Instance>(
```

Comment:
Updated version of the approach described here https://ollylut.medium.com/what-is-useforkref-hook-4be1c85d2d1b

Not sure what to do with the warning below if anyone has ideas